### PR TITLE
Feat: Added Bulk Edit feature 

### PR DESCRIPTION
**Pull request by Abhishek9503**
## Description

#1008 
#1009 
In this PR i have added both Bulk Edit and Bulk Paramters feature

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build

## Have you tested locally?

- [x] 👍 yes
- [ ] 🙅 no, because I am lazy

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md

### CODE REVIEW

File: src/packages/@app/styles/style.scss [hunk 1]
```diff
@@ -259,6 +260,7 @@ $theme-red-230408: #230408;
   --text-secondary-300 : #{$theme-grey-45494D};
   --text-secondary-350 : #{$theme-grey-636566};
   --text-secondary-400 : #{$theme-grey-313233};
+  --text-secondary-550 : #{$theme-grey-808080};
```
Comment (Astitva877) [inline src/packages/@app/styles/style.scss:263]: please follow the numbering structure here.

File: src/packages/@app/database/database.ts [hunk 2]
```diff
@@ -176,6 +176,9 @@ export class RxDB {
           8: function (oldDoc: TabDocument) {
             return oldDoc;
           },
+          9: function (oldDoc: TabDocument) {
+            return oldDoc;
+          },
```
Comment (itsmdasifraza) [inline src/packages/@app/database/database.ts:181]: add migrated tab doc fields here

File: src/packages/@common/types/workspace/request.ts [hunk 3]
```diff
@@ -281,11 +310,13 @@ export interface AuthWrapper {
 
 export interface HeadersWrapper {
   headers: KeyValueChecked[];
+  isBulkEditEnabled: boolean;
```
Comment (itsmdasifraza) [inline src/packages/@common/types/workspace/request.ts:313]: not required

**Review by itsmdasifraza — CHANGES_REQUESTED** [review]

File: src/packages/@app/database/database.ts (hunk 2 above)
Comment (Abhishek9503) [reply src/packages/@app/database/database.ts:181]: I didn't got your point....

File: src/packages/@common/types/workspace/request.ts [hunk 4]
```diff
@@ -137,6 +137,14 @@ export interface IsSaveDescriptionInProgressWrapper {
 export interface IsSaveRequestInProgressWrapper {
   isSaveRequestInProgress: boolean;
 }
+export interface IsBulkHeaderWrapper {
+  isBulkHeader: boolean;
+}
+export interface IsBulkParameterWrapper {
+  isBulkParameter: boolean;
+}
+
```
Comment (itsmdasifraza) [inline src/packages/@common/types/workspace/request.ts:146]: update here also

File: src/packages/@workspaces/features/rest-explorer/layout/RestExplorer.svelte [hunk 5]
```diff
@@ -313,6 +316,9 @@
                 />
               {:else if $tab.property.request?.state?.requestNavigation === RequestSectionEnum.HEADERS}
                 <RequestHeaders
+                isBulkEditActive={$tab?.property?.request.state
+                  ?.isBulkHeader}
```
Comment (itsmdasifraza) [inline src/packages/@workspaces/features/rest-explorer/layout/RestExplorer.svelte:320]:
```
isBulkEditActive={$tab?.property?.request.state
                  ?.isHeaderBulkEditActive}
```

**Comment by itsmdasifraza** [comment]
1 comment is still unresolved for "switch generic component"

**Comment by Abhishek9503** [comment]
> 1 comment is still unresolved for "switch generic component"

Done

**Review by itsmdasifraza — APPROVED** [review]

**Review by gc-codes — APPROVED** [review]

**Pull request merged**